router/usecase/pools/routable_cw_alloy_transmuter_pool_test.go

```diff
@@ -0,0 +1,275 @@
+								{Denom: ALLUSD, NormalizationFactor: osmomath.NewInt(10)},
+							},
+						},
+					},
+				),
+				PoolType: poolType,
+			}
+
+			routablePool, err := pools.NewRoutablePool(mock, USDT, noTakerFee, domain.CosmWasmPoolRouterConfig{
+				IsAlloyedTransmuterEnabled: true,
+			})
+			s.Require().NoError(err)
+
+			r := routablePool.(*pools.RouteableAlloyTransmuterPoolImpl)
+
+			inNormFactor, outNormFactor, err := r.FindNormalizationFactors(tc.tokenInDenom, tc.tokenOutDenom)
+
+			if tc.expectError != nil {
+				s.Require().Error(err)
+				s.Require().ErrorIs(err, tc.expectError)
+			} else {
+				s.Require().NoError(err)
+				s.Require().Equal(tc.expectedInNormFactor, inNormFactor)
+				s.Require().Equal(tc.expectedOutNormFactor, outNormFactor)
+			}
+		})
+	}
+}
+
+func (s *RoutablePoolTestSuite) TestCalcTokenOutAmt_AlloyTransmuter() {
+	tests := map[string]struct {
+		tokenIn          sdk.Coin
+		tokenOutDenom    string
+		expectedTokenOut osmomath.BigDec
+		expectedError    error
+	}{
+		"valid calculation using normalization factors": {
+			tokenIn:          sdk.NewCoin(USDC, osmomath.NewInt(100)),
+			tokenOutDenom:    USDT,
+			expectedTokenOut: osmomath.NewBigDec(1), // (100 * 1) / 100 = 1
+			expectedError:    nil,
+		},
+		"valid calculation with decimal points": {
+			tokenIn:          sdk.NewCoin(USDC, osmomath.NewInt(10)),
+			tokenOutDenom:    USDT,
+			expectedTokenOut: osmomath.MustNewBigDecFromStr("0.1"), // (10 * 1) / 100 = 0.1
+			expectedError:    nil,
+		},
+		"valid calculation, truncated to zero": {
+			tokenIn:          sdk.NewCoin("overlypreciseusd", osmomath.NewInt(10)),
+			tokenOutDenom:    USDC,
+			expectedTokenOut: osmomath.MustNewBigDecFromStr("0"),
+			expectedError:    nil,
+		},
+		"missing normalization factor for token in": {
+			tokenIn:          sdk.NewCoin("INVALID", osmomath.NewInt(100)),
+			tokenOutDenom:    USDT,
+			expectedTokenOut: osmomath.ZeroBigDec(),
+			expectedError:    domain.MissingNormalizationFactorError{Denom: "INVALID", PoolId: 1},
+		},
+		"missing normalization factor for token out": {
+			tokenIn:          sdk.NewCoin(USDC, osmomath.NewInt(100)),
+			tokenOutDenom:    "INVALID",
+			expectedTokenOut: osmomath.ZeroBigDec(),
+			expectedError:    domain.MissingNormalizationFactorError{Denom: "INVALID", PoolId: 1},
+		},
+		"missing normalization factors for both token in and token out": {
+			tokenIn:          sdk.NewCoin("INVALID", osmomath.NewInt(100)),
+			tokenOutDenom:    "INVALID",
+			expectedTokenOut: osmomath.ZeroBigDec(),
+			expectedError:    domain.MissingNormalizationFactorError{Denom: "INVALID", PoolId: 1},
+		},
+	}
+
+	for name, tc := range tests {
+		s.Run(name, func() {
+			s.Setup()
+
+			cosmwasmPool := s.PrepareCustomTransmuterPool(s.TestAccs[0], []string{USDC, USDT})
+
+			poolType := cosmwasmPool.GetType()
+
+			veryBigNormalizationFactor, _ := osmomath.NewIntFromString("999999999999999999999999999999999999999999")
```

Comment:
nit: can we define this at the top of the test fixture to avoid reinitializing for every single one